### PR TITLE
Compiles under VS2015 now, with the changes to Vorb.

### PR DESCRIPTION
**Pull request by caseymcc**
This is not complete, same as the Vorb PR. But it does compile and run, hover I need to get FreeImage working for it to really run.

Vorb PR - https://github.com/RegrowthStudios/Vorb/pull/99, I set it up so that Vorb should be along side the Soa directory and used add_subdirectory in cmake so that it part of the Soa build making it easier to modify and debug with Soa.

### CODE REVIEW

**Comment by Barnold1953** [comment]
Thank you for switching to GLM!

File: SoA/BlockTexture.h [hunk 1]
```diff
@@ -114,17 +118,17 @@ class BlockTextureLayer {
 
     void getFinalColor(OUT color3& color, ui8 temperature, ui8 rainfall, ui32 altColor) const;
 
-    ConnectedTextureMethods method = ConnectedTextureMethods::NONE;
-    ui8v2 size = ui8v2(1);
-    ConnectedTextureSymmetry symmetry = ConnectedTextureSymmetry::NONE;
-    ConnectedTextureReducedMethod reducedMethod = ConnectedTextureReducedMethod::NONE;
-    BlockColorMap* colorMap = nullptr;
-    color3 averageColor = color3(255, 255, 255); // Average texture color combined with color (for terrain)
-    color3 color = color3(255, 255, 255);
```
Comment (Barnold1953) [inline SoA/BlockTexture.h:123]: What is wrong with the inline initialization?

**Comment by Barnold1953** [comment]
Is it a compile reason or a design reason to remove all of the header initialization of variables and moving it to constructors?

File: SoA/BlockTexture.h (hunk 1 above)
Comment (caseymcc) [reply SoA/BlockTexture.h:123]: per standard some of the implicit default constructors get deleted (mainly the copy constructor). This leaves the code un-compilable by compiler (at least by VS2015)

**Comment by caseymcc** [comment]
Not sure I would have merge these PRs just yet. I have more changes coming (just finished hacking in png lib in place of FreeImage in Vorb). Currently I can run SoA up to the developer menu, some test work but there also seems to be some missing files in the SoAGameData. Can't run the SoA fully though as there is a crash in Lua that I am feebly trying to find. Hunter by default does not support debugging outside libs very well so trying to work through that.

**Comment by Barnold1953** [comment]
Ah I see, just make a new pull request when you are ready. The branch isn't exactly stable anyways. I can help you figure out the Lua crash if I can get it to compile.

**Comment by caseymcc** [comment]
The way it is setup now you should just need to git SoA, then git Vorb inside the SoA directory (maybe turn into sub module) along side deps, games, SoA, etc... Run cmake at the top level, I use cmake -H. -B_buildx64 -G "Visual Studio 14 2015 Win64"

From there hunter should fetch and compile everything needed, no need for the Deps repos.

**Comment by caseymcc** [comment]
I created a discord server for SoA if you want to join, should be a better place to go back and forth.

https://discord.gg/PkUvfm

**Comment by caseymcc** [comment]
Its alive, https://youtu.be/-rV4VobfG8Q

**Pull request merged**